[READY FOR TM] Adds the Brig Physician

SgtHunk opened the pull request:
## About The Pull Request

This adds the Brig Physician/Doctor as one more of the security roles for the station crew. They're mindshielded, and can't be most antagonists. This gives security a Doctor they can trust that will help keep the Officers/Prisoners alive!
Their equipment consists of:

- A medical armored beret, the same one deputies get.
- Their own special armored labcoat, with slightly lower stats than a normal armor vest, but added Bio protection.
- A bottle of Pax pills to pacify those pesky prisoners for a while.
- His own personal defense weapon, a medical baton in case things get aggressive. Although weaker than a normal stun/telescopic baton, it's ability to apply electrical shocks makes it useful for revivals.
- A regular first-aid kit.
- Medical sunglasses, to identify how hurt Officers and Prisoners are.
- A medical belt, filled with standard surgery tools.

![image](https://user-images.githubusercontent.com/68669754/90948684-19fee900-e40f-11ea-9436-bb15435ca37d.png)
![image](https://user-images.githubusercontent.com/68669754/91362090-ab7fa980-e7c7-11ea-90f7-da25ba305132.png)
![image](https://user-images.githubusercontent.com/68669754/91504770-10a4cf00-e89c-11ea-850e-5834a6c992b1.png)

Lizards get their variation of a Digisuit. Fun.
Added support for Digitigrade lizards, giving them their jackboots once they join.
Added support for Plasmemes too, their own envirosuit and helmet.
DISCLAIMER: Beefmen (due to their nature) don't get a labcoat, and instead just get their Security sash.

Credits to:

- Pengu for the Armored labcoat sprite
- MaltVinegar for the HUD/Card sprites
- Guill for pointing me into the right direction regarding landmarks
- #163  for the Brigdoc jumpsuit sprites  
- https://github.com/Skyrat-SS13/Skyrat13/pull/933 for the Envirosuit sprites

To do:

- [x]  Fix Brig Physician/Deputy not having designated land marks, making them spawn on arrivals shuttle. (Help would be appreciated)

## Why It's Good For The Game

Less messing around the HOP because someone wants to be a Brig Doctor, people will be able to become one and Security will be able to trust them as they can't be antagonists. 
Also (hopefully) fewer prisoners/brigged people will die because of security harmbatoning them to crit. Woo.

## Changelog
:cl:
add: Added the Brig Physician, with armored labcoat, medical baton and everything. (Even plasmeme support)
fix: Fixed deputy being able to become antag.
imageadd: added Brig Physician HUD/ID icons.
spriteadd: added the Brig Doc Jumpsuit, Envirosuit, Envirohelmet and Labcoat.
code: Changed some of the deputy/fulp proc code, defining fulp jobs in case anyone is interested in adding more.
tweak: tweaked the stun baton's revival surgery chance, to make the medical baton more viable.
/:cl:

SgtHunk committed:
Most of the brig doc stuff + more

Adds Brig Physician as a job.
Excludes Deputy + Brig Physician from antags.
Adds Functional ID and HUD icons for Brig Physician.
Adds the armored labcoat, exclusive to the Brig Physician.
Reworks a proc, and adds the Brig Physician heirloom's.

SgtHunk committed:
Gives the Brig Doctor a surgery-prepared belt, they get a regular first-aid on hand instead

SgtHunk committed:
Brig Physician Jumpsuit

Gives Brig Physician a special Jumpsuit/Jumpskirt, using some of the left-over sprites that Mith made

Squishypone commented:
idk I kind of like the fact that deputies can become a antag

SgtHunk commented:
I spoke with the ex-headmin, and he did say that Deputies being antag was a bug. This just wraps it up so that it's guaranteed that they're not antag.

ATH1909 commented:
This kind of seems redundant with sec officers and deputies who've been assigned to Medbay.

I suppose that I can see the niche for them, though. Will they be allowed to leave the Brig and/or hunt antagonists as a normal officer would be able to?

SgtHunk commented:
Their main purpose will be staying on the brig to heal up wounded/dead officers and prisoners, aid in the case of checking for implants/larvae/any hidden stuff through surgery, _rehabilitate_ pacify patients, and let security have a doctor they can trust will add that Anti-drop implant into them rather than removing their brain. They could also help against most antags using their medical expertise. Hell, with IC reasons, they can identify bloodsuckers. But they're mainly there to stay in the brig, go out once in a while to get more healing items, from anti-toxins to deal with that poisoned sec officer, a vaccine to a virus, get implants from the med techfab, go get stasis beds/defib mounts/belt defibs if they feel like it, etc.
If HOS (With Cap's permission) allows it, they could even be the ones carrying out the lethal injection. _But that's more RP territory_
In a best-case scenario, they stay on the brig, give the warden some company, and make sure the officers/prisoners/brigged criminals are in good conditions. They shouldn't be hunting antags most of the time, that's what officers are for. Although, that baton exists for a reason.

SgtHunk committed:
FUCK

fixes icons breaking for literally every job other than Brig Doc and Deputy, still need to figure out landmarks
Oh yeah and also Brig Doc gets a bottle of formaldehyde

BarrelOfNapkins commented:
I like everything but the telescopic baton; this item is super strong. Why not a flash?

SgtHunk commented:
I do agree that the telebaton is quite strong at times, it was the best I could find close to self-defense weapon. 
They already get a generous amount of access for a flash, or even a stun baton if they bother enough. My other idea was maybe making a subtype of a contractor/telebaton, but with weaker stam damage, long cooldown and little knockdown. But that would need some spriting

BarrelOfNapkins commented:
I guess next question I have is why would a brig doctor need a self defense weapon? Anytime im HOS and hire a doctor, I only arm them if I find they need it.
Also BTW changeling is for players, they don't care what procs or code was changed

SgtHunk commented:
Guess for a similar reason Heads get a telebaton. I can make a compromise and try to add a weaker baton for the brig doctor, something like 40 stam damage with a 3 second cooldown and a bit of knockdown? Should make it a weak-enough baton while still keeping it slightly useful for stuff like self-defense and fending off prisoners/brigged people. Bringing it to the longest timer for stamcritting someone (Stun baton being 2ish, Telebaton being 4 and with this being 6)
Or maybe that feels like too big of a downgrade so the doctor will just ask for a normal stun baton.

Squishypone commented:
Maybe instead of a baton have a syringe gun with some pax or something like that

SgtHunk commented:
I'm giving them their own baton as mentioned above, will also have an utility function so it's not simply a baton downgrade, and will actually serve a purpose on the Brig Doc's equipment

SgtHunk committed:
Code stuff

Moves all the sprites to a new folder called enricode_icons, also changes a bit of surgery to nerf the stun baton.
Oh yeah, also Brig Doctors get their own 'medical baton' which is a nerfed telebaton. And not only does it serve self-defense purposes, it also has an utility purpose due to its ability to apply electrical shocks.

SgtHunk committed:
Finishes the Pull

Adds Skyrat's Brig Doc plasmeme envirosuits and helmets, also actually has the landmark added

SgtHunk committed:
FUCK part 2

probably stops the sec officer landmark from being qdeleted (hopefully, local testing is tough)

Squishypone commented:
maybe instead of the normal penlight you can give the paramedic one because it has a kind of blinding effect because they don't have a flash

SgtHunk commented:
They can easily get a flash, they still have a decent amount of access.

Squishypone commented:
looks like a lot of fun can't wait for it to be TM

SgtHunk committed:
Fixes Digisuit + Sash

Moves Digisuit to Mith's stash (because I'm not gonna bother adding more code)
Also Beef Physicians get Sec stash on arrival, woo

SgtHunk commented:
github bot have mercy

Squishypone commented:
Can this get test merged anytime soon I love this idea

SgtHunk commented:
Considering how wind is moving us to a 2nd repo, it will probably have to get TM'd when we get it going and move our PRs to that repo

JohnFulpWillard commented on code/__DEFINES/jobs.dm:
```diff
@@ -44,4 +44,5 @@
 #define JOB_DISPLAY_ORDER_DETECTIVE 33
 #define JOB_DISPLAY_ORDER_SECURITY_OFFICER 34
-#define JOB_DISPLAY_ORDER_PRISONER 35
+#define JOB_dISPLAY_ORDER_BRIG_PHYSICIAN 35
```
i cant fucking believe the d isn't capitalized my fucking IMMERSION

SgtHunk committed:
flexes on john

he yelled at me for a mistake so i deleted that entire line, get fucked

Squishypone commented:
sad I really like this idea

SgtHunk commented:
Just wait for 2nd repo, I still have the branch
